### ISSUE TITLE
Crash when declaring external function with array of struct that possesses arrays

### ISSUE THREAD

**Issue by mattaereal** [ethereum/solidity#4713]
_This issue is being reported as part of the current audit being held by Zeppelin Solutions to Solidity's compiler (tag v0.4.24)._

## Description
Compiler crashes using `pragma experimental ABIEncoderV2`, when an array of structs that is composed by one or more arrays is used as a parameter on an external function of a library.

## Error in file
Empty message.
`libsolidity/codegen/ABIFunctions.cpp(1191)`

## Displayed message
`Throw in function std::__cxx11::string dev::solidity::ABIFunctions::abiDecodingFunctionCalldataArray(const dev::solidity::ArrayType&)`

## Example code
```solidity
pragma experimental ABIEncoderV2;                                                                                                              
pragma solidity ^0.4.24;                                                                                                                       
                                                                                                                                              
library Test {                                                                                                                                 
   struct Nested { int[] a; }                                                                                                                 
   function Y(Nested[]) external {}                                                                                                           
}
```

**Comment by axic** [comment]
Reproducible in 0.4.24 and still happens on develop (albeit has a nicer message):
```
Unimplemented feature:
/Users/alex/Projects/solidity/libsolidity/codegen/ABIFunctions.cpp(1191): Throw in function std::__1::string dev::solidity::ABIFunctions::abiDecodingFunctionCalldataArray(const dev::solidity::ArrayType &)
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::UnimplementedFeatureError>
std::exception::what: Calldata arrays with non-value base types are not yet supported by Solidity.
[dev::tag_comment*] = Calldata arrays with non-value base types are not yet supported by Solidity.
```

**Comment by christianparpart** [comment]
@axic I just tested it right now with `develop`. and It's compiling fine now.

**Comment by christianparpart** [comment]
@axic I've been bisecting this, and now it becomes clear. It got fixed in via #4738, i.e. we enforce a data location in function args now, so this bug cannot occur anymore. Should we close this issue then?

**Comment by leonardoalt** [comment]
Closing since the crash is fixed.
